Kinder/Settings.xaml.cs

```diff
@@ -27,47 +27,60 @@ public Settings()
 
         private void ChangeEmail_Click(object sender, RoutedEventArgs e)
         {
-            if (RegValidation.CheckIfEmailValid(NewEmail.Text))
+            try
             {
+                RegValidation.CheckIfEmailValid(NewEmail.Text);
                 User.ChangeUserEmail(NewEmail.Text);
                 MessageBox.Show("Email changed to " + NewEmail.Text + ".");
```

Comment:
wouldn't user see message "email changed to ...", if their email was not valid?

Reply:
hmm, or maybe code doesn't reach this line, when exception is thrown

Reply:
When an exception is thrown, it does not go to following code, it goes straight to catch block